### ISSUE TITLE
[Bug]: Ticklabel padding different between top and bottom

### ISSUE THREAD

**Issue by baloe** [matplotlib/matplotlib#22800]
### Bug summary

The vertical padding between ticks and ticklabels above an x-axis is too large.
One could tweak the `xtick.major.pad` setting, but then the ticklabels below an x-axis are off.

### Code for reproduction

```python
import matplotlib.pyplot as plt

fig, ax = plt.subplots()

ax.tick_params( top=True, labeltop=True, labelright=True, right=True )

fig.savefig('test.pdf')
```

### Actual outcome

output, with side-by-side comparison of ticklabel spacings.

![test](https://user-images.githubusercontent.com/12622691/162329792-993fe02a-9b3c-41c5-98fc-2adefd4d25c6.png)

### Expected outcome

Same spacing between ticks and ticklabels on all four sides.

### Additional information

I find this more noticeable when reducing the font size, which is something I do when intending to embed plots in a print publication.

```python
import matplotlib.pyplot as plt
from matplotlib import rc

rc('xtick',  labelsize=7 )
rc('ytick',  labelsize=7 )

fig, ax = plt.subplots()
ax.tick_params( top=True, labeltop=True, labelright=True, right=True )
fig.savefig('test2.png')
```
![test2](https://user-images.githubusercontent.com/12622691/162331021-fc65376a-1255-43d7-b9df-a73cdbf33676.png)

### Matplotlib Version

3.5.1

### Matplotlib Backend

QtAgg

### Python version

3.8.13

### Installation

conda

**Comment by QuLogic** [comment]
The vertical alignment for the top ticks is 'bottom', not 'baseline'. This leaves space for the descenders, which show up mostly in general characters and not numbers. I don't think there is an easy way to override this alignment.

**Comment by anntzer** [comment]
It would seem reasonable to switch this to "baseline" (even in that case, the normal pad will often be enough even for ticklabels that actually contain descenders).

**Comment by baloe** [comment]
As a workaround for a case where an axis only has top ticks: Is there a variable that will tell me the height of the baseline relative to the bottom so that I can compensate for that?

**Comment by QuLogic** [comment]
This is dependent on fonts and I'm not sure there's an easy way to determine that. At a lower level, this is determined by the result of `Renderer.get_text_width_height_descent` using a string "lp" and the relevant font settings. At a higher level, you may be able to create a `Text` with 'l' and 'p', and then compare `Text.get_window_extent` to see the size change.

**Comment by timhoffm** [comment]
Agreed that baseline would be the slightly preferred choice when doing this from scratch. We have three options:

- do a hard chance. I‘m suppose we are not guaranteeing pixel-perfect image stability. But this  will cause some of our baseline image tests to fail.
- Add a control flag either an internal variable or a rcParam. This would make the change a little less hard, but still there is no good deprecation path either a runtime notification.
- decide it’s not worth the effort and do nothing.

I don’t have a strong preference.
